src/client/components/MyProfile/ProfilePage/ProfileComponent.js

```diff
@@ -0,0 +1,80 @@
+import { useAuthentication } from '../../../hooks/useAuthentication';
+import React, { useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import './ProfileComponent.styles.css';
+import { ProfileCardComponent } from '../../ProfileCardComponent/ProfileCardComponent';
+import { YourUploadedCVs } from '../YourUploadedCVs/YourUploadedCVs';
+import TitleDesc from '../../Title/TitleDesc';
+import { SentReviewsComponent } from '../../SentReviewsComponent/SentReviewsComponent.js';
+
+export default function ProfileComponent() {
+  const { userData } = useAuthentication();
+  const history = useHistory();
+  const [cvsList, setCvsList] = useState([]);
+  const [reviews, setReviews] = useState([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [reviewsLoaded, setReviewsLoaded] = useState(false);
+  useEffect(() => {
+    if (userData) {
+      (async () => {
+        try {
+          const response = await fetch('/api/cvs');
+          if (response.status !== 200) {
+            throw new Error('fail to connect to the api/cvs');
+          }
+          const userCvs = await response.json();
+          setIsLoaded(true);
+          if (userCvs) {
+            setCvsList(() =>
+              userCvs
+                .filter((cv) => cv.fk_user_id === userData.id)
+                .map((cv) => {
+                  return { ...cv, createdDate: cv.createdAt };
+                }),
+            );
+          }
+          const responseRew = await fetch(`api/reviews/${userData.id}`);
+          if (responseRew.status !== 200) {
+            throw new Error('fail to connect to the api/reviews');
+          }
+          const reviewsCv = await responseRew.json();
+          setReviewsLoaded(true);
+          if (reviewsCv) {
+            setReviews(reviewsCv);
+          }
+        } catch (error) {
+          // if data not found in api/usercv
+          history.push('/sign-in');
```

Comment:
Why are we sending the user to `sign in`  if ther API errors out?

Reply:
Thanks @bytewiz Removed.